routes/spotifyRouter.js

```diff
@@ -20,30 +21,48 @@ router.get('/', function (req, res, next) {
 });
 
 router.get('/login', (req, res) => {
-    let spotifyApi = req.app.locals.main.spotifyApi
+
+    let main = req.app.locals.main;
+
+    if (!main || !main.spotifyApi) {
```

Comment:
Needed to do two assignments because the old main would still be referring to the old null/undefined!

Reply:
Nice catch, can you check what happens if we just do `main = new Main()` and if that changes req.app.locals.main?

Reply:
Checked and nope this doesn't because we're not referencing back!

Reply:
Ahh got it okay nice